tests/subsys/audio_modules/audio_module_template/src/template_test.c

```diff
@@ -0,0 +1,311 @@
+			  "Failed open for names, instance name should be %s, but is %s",
+			  test_instance_name, handle.name);
+	zassert_equal(handle.state, AUDIO_MODULE_STATE_CONFIGURED,
+		      "Not in configured state after call to open the audio module");
+
+	ret = audio_module_names_get(&handle, &base_name, &instance_name[0]);
+	zassert_equal(ret, 0, "Get names function did not return successfully (0): ret %d", ret);
+	zassert_mem_equal(base_name, audio_module_template_description->name,
+			  strlen(audio_module_template_description->name),
+			  "Failed get names, base name should be %s, but is %s",
+			  audio_module_template_description->name, base_name);
+	zassert_mem_equal(&instance_name[0], test_instance_name, strlen(test_instance_name),
+			  "Failed get names, instance name should be %s, but is %s", handle.name,
+			  &instance_name[0]);
+
+	ret = audio_module_configuration_get(
+		&handle, (struct audio_module_configuration *)&return_configuration);
+	zassert_equal(ret, 0, "Configuration get function did not return successfully (0): ret %d",
+		      ret);
+	zassert_equal(return_configuration.bit_depth, configuration.bit_depth,
+		      "Failed to get configuration");
+	zassert_equal(return_configuration.sample_rate_hz, configuration.sample_rate_hz,
+		      "Failed to get configuration");
+	zassert_mem_equal(return_configuration.some_text, configuration.some_text,
+			  sizeof(CONFIG_TEXT), "Failed to get configuration");
+
+	ret = audio_module_connect(&handle, NULL, true);
+	zassert_equal(ret, 0, "Connect function did not return successfully (0): ret %d", ret);
+	zassert_equal(handle.use_tx_queue, 1, "Connect queue size is not %d, but is %d", 1,
+		      handle.use_tx_queue);
+
+	ret = audio_module_start(&handle);
+	zassert_equal(ret, 0, "Start function did not return successfully (0): ret %d", ret);
+	zassert_equal(handle.state, AUDIO_MODULE_STATE_RUNNING,
+		      "Not in running state after call to start the audio module");
+
+	for (i = 0; i < TEST_AUDIO_DATA_ITEMS_NUM; i++) {
+		audio_data_tx.data = (void *)&test_data_in[i * TEST_MOD_DATA_SIZE];
+		audio_data_tx.data_size = TEST_MOD_DATA_SIZE;
+		memcpy(&audio_data_tx.meta, &test_metadata, sizeof(struct audio_metadata));
+
+		audio_data_rx.data = (void *)&test_data_out[i * TEST_MOD_DATA_SIZE];
+		audio_data_rx.data_size = TEST_MOD_DATA_SIZE;
+
+		ret = audio_module_data_tx_rx(&handle, &handle, &audio_data_tx, &audio_data_rx,
+					      TEST_TX_RX_TIMEOUT_US);
+		zassert_equal(ret, 0, "Data TX-RX function did not return successfully (0): ret %d",
+			      ret);
+		zassert_mem_equal(audio_data_tx.data, audio_data_rx.data, TEST_MOD_DATA_SIZE,
+				  "Failed to process data");
+		zassert_equal(audio_data_tx.data_size, audio_data_rx.data_size,
+			      "Failed to process data, sizes differ");
+		zassert_mem_equal(&audio_data_tx.meta, &audio_data_rx.meta,
+				  sizeof(struct audio_metadata),
+				  "Failed to process data, meta data differs");
+
+		if (i == TEST_AUDIO_DATA_ITEMS_NUM / 2) {
```

Comment:
I dont quite get why we need this `if`

Reply:
It is a point in the flow of data that the module is reconfigured and then it continues with the new configuration. Thus testing a reconfiguration on a running stream.